src/main/java/edu/tamu/iiif/service/MessageListenerService.java

```diff
@@ -0,0 +1,66 @@
+package edu.tamu.iiif.service;
+
+import static edu.tamu.iiif.utility.StringUtility.encode;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import edu.tamu.iiif.controller.ManifestRequest;
+import edu.tamu.iiif.exception.InvalidUrlException;
+import edu.tamu.iiif.exception.UnknownActionException;
+import edu.tamu.iiif.model.RedisManifest;
+import edu.tamu.iiif.model.repo.RedisManifestRepo;
+import edu.tamu.weaver.messaging.annotation.WeaverMessageListener;
+
+@Service
+public class MessageListenerService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MessageListenerService.class);
+
+    @Autowired
+    private RedisManifestRepo manifestRepo;
+
+    @Autowired
+    private List<ManifestService> manifestServices;
+
+    @WeaverMessageListener(destination = "${iiif.messaging.channel}", containerFactory = "topicContainerFactory")
+    private void update(Map<String, String> message) throws UnknownActionException, InvalidUrlException {
+        switch (message.get("action")) {
+        case "METADATA_CREATE":
+        case "METADATA_UPDATE":
+        case "METADATA_DELETE":
+            updateManifest(message.get("contextPath"), message.get("repositoryType"));
+            break;
+        case "RESOURCE_CREATE":
+        case "RESOURCE_DELETE":
+            updateManifest(message.get("parentContextPath"), message.get("repositoryType"));
+            break;
+        default:
+            throw new UnknownActionException("The action " + message.get("action") + "is not defined for this service");
```

Comment:
There is nothing that is going to handle this exception. It is not needed. A log will suffice.

Messages that come through this channel that should be ignored by this service is not an exception. The log is needed to determine if the channel subscription is be saturated with unnecessary messages. If that is the case we need to be more granular about the topics and their subscription/publication.

Reply:
https://github.com/TAMULib/IRIIIFService/commit/b48dea90a4271025c9772504f717734f25893795